src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderAdapter.cs

```diff
@@ -18,15 +19,12 @@ namespace NuGet.PackageManagement.VisualStudio
     public class VsCredentialProviderAdapter : ICredentialProvider
     {
         private readonly IVsCredentialProvider _provider;
+        private readonly IVsSolutionManager _solutionManager;
 
-        public VsCredentialProviderAdapter(IVsCredentialProvider provider)
+        public VsCredentialProviderAdapter(IVsCredentialProvider provider, IVsSolutionManager solutionManager)
```

Comment:
I am just curious on passing `VsShellUtilities.ShutdownToken` instead of adding a dependency on `IVsSolutionManager`. This is definitely not a blocking comment.

Reply:
Doing that broke our tests. See the linked PRs in the PR description.